fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricBlockStateDefinitionProvider.java

```diff
@@ -16,26 +16,14 @@
 
 package net.fabricmc.fabric.api.datagen.v1.provider;
 
-import net.minecraft.data.client.BlockStateDefinitionProvider;
-import net.minecraft.data.client.ItemModelGenerator;
-import net.minecraft.data.client.model.BlockStateModelGenerator;
-
 import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
 
 /**
- * Extend this class and implement {@link FabricBlockStateDefinitionProvider#generateBlockStateModels} and {@link FabricBlockStateDefinitionProvider#generateItemModels}.
- *
- * <p>Register an instance of the class with {@link FabricDataGenerator#addProvider} in a {@link net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint}
+ * @deprecated use {@link FabricModelProvider} instead.
  */
-public abstract class FabricBlockStateDefinitionProvider extends BlockStateDefinitionProvider {
-	protected final FabricDataGenerator dataGenerator;
```

Comment:
Removing this will break backwards compat.

Reply:
Assuming this is some binary compatibility thing, as these fields are still available in the non-deprecated superclasses. Added back in latest commit.